requirements.txt

```
@@ -1,10 +1,10 @@
 peft @ git+https://github.com/huggingface/peft.git
 transformers @ git+https://github.com/huggingface/transformers.git
+bitsandbytes @ git+https://github.com/TimDettmers/bitsandbytes.git
```

`pip install bitsandbytes>=0.39.0` based on his new PR

The main repo is `38.1`. I am not sure how his main repo version is older than the required version.

Ref: https://github.com/huggingface/transformers/blob/0b050922b70fd9864b4b751e83f02b7f7cbd9e2d/docs/source/en/main_classes/quantization.mdx

```suggestion
bitsandbytes>=0.39.0
```